enterprise/api/v1/views.py

```diff
@@ -64,6 +64,54 @@ class EnterpriseCustomerViewSet(EnterpriseReadOnlyModelViewSet):
     filter_fields = FIELDS
     ordering_fields = FIELDS
 
+    @detail_route()
+    def courses(self, request, pk=None):  # pylint: disable=invalid-name
+        """
+        Retrieve the list of courses contained within the catalog linked to this enterprise.
+
+        Only courses with active course runs are returned. A course run is considered active if it is currently
+        open for enrollment, or will open in the future.
+        ---
+        serializer: serializers.CourseSerializerExcludingClosedRuns
+        """
+        enterprise_customer = self.get_object()
+
+        # Make sure there is a catalog associated with the enterprise
+        if not enterprise_customer.catalog:
+            error_message = ("No catalog is associated with the given enterprise from endpoint "
+                             "'/enterprise-customer/{}/courses'.".format(pk))
+            logger.error(error_message)
+            raise NotFound("The resource you are looking for does not exist: " + error_message)
+
+        # Ensure that the user making this request is either a staff user or is linked to the enterprise
+        if not request.user.is_staff:
+            try:
+                models.EnterpriseCustomerUser.objects.get(
+                    enterprise_customer=enterprise_customer,
+                    user_id=request.user.id,
+                )
+            except models.EnterpriseCustomerUser.DoesNotExist:
+                error_message = ("User '{}' is not associated with enterprise from endpoint "
+                                 "'/enterprise-customer/{}/courses'.".format(request.user.username, pk))
+                logger.error(error_message)
+                raise PermissionDenied("The user does not have permission to access this resource: " + error_message)
+
+        catalog_api = CourseCatalogApiClient(request.user)
+        courses = catalog_api.get_catalog_courses(enterprise_customer.catalog)
+
+        # An empty response can mean that there was a problem fetching data from catalog API.
+        if not courses:
```

Comment:
I don't think we should do an error here. It's not semantically meaningful to the API consumer to distinguish between _reasons_ the EC doesn't have any courses in its catalog, and in fact, this could produce an error in cases where the EnterpriseCustomer actually does have an empty catalog.